src/net_processing.cpp

```diff
@@ -2039,7 +2076,16 @@ void PeerManagerImpl::NewPoWValidBlock(const CBlockIndex *pindex, const std::sha
 void PeerManagerImpl::UpdatedBlockTip(const CBlockIndex *pindexNew, const CBlockIndex *pindexFork, bool fInitialDownload)
 {
     SetBestHeight(pindexNew->nHeight);
-    SetServiceFlagsIBDCache(!fInitialDownload);
+
+    // Update internal sync state.
+    // Avoid using 'fInitialDownload' alone since the flag currently cannot go backwards after passing IBD once.
+    // This means that, after completing sync for the first time, 'fInitialDownload' will remain true even
+    // when the node has been stalled for days and the arriving block is within the IBD window.
+    if (!fInitialDownload && IsBlockCloseToTip(pindexNew, m_chainparams.GetConsensus())) {
```

Comment:
> Seems like `!fInitialDownload && ` can be removed? If `IsBlockCloseToTip()` is `true` then that block is 3h20min or younger. In this case `fInitialDownload` will always be `false` because it uses 24h, right?

It is a simple and quick check to not execute `IsBlockCloseToTip()` during IBD at all.